### PR TITLE
 #162 Switch to license expression

### LINKED ISSUE #162: Nuget Package should be using a license expression
### Summary
I wish for the nuget packages to have the licence expression property set correctly

### Details

The licence expression property should be set to the correct licence type I.e. Apache-2.0 as this will enable analysis of licences in use to occur in external tools & the license type will be shown in Nuget etc.

### PR DESCRIPTION
**Pull request by thompson-tomo**
Closes #162

### CODE REVIEW

File: csharp/Std.UriTemplate.csproj [hunk 1]
```diff
@@ -19,7 +19,7 @@
     <RepositoryUrl>https://github.com/std-uritemplate/std-uritemplate</RepositoryUrl>
     <IncludeSymbols>true</IncludeSymbols>
     <SymbolPackageFormat>snupkg</SymbolPackageFormat>
-    <PackageLicenseFile>LICENSE</PackageLicenseFile>
```
Comment (andreaTP) [inline csharp/Std.UriTemplate.csproj:22]: Do we need to remove this line?

**Comment by andreaTP** [comment]
Thanks for making the changes, other than keeping the reference to the file lgtm.

File: csharp/Std.UriTemplate.csproj (hunk 1 above)
Comment (thompson-tomo) [reply csharp/Std.UriTemplate.csproj:22]: Na, it is fine to leave it in as it results in the license file also being included in the package.

**Comment by thompson-tomo** [comment]
Keeping the reference will result in the license file also being included in the package.

**Comment by andreaTP** [comment]
thanks for reverting back @thompson-tomo 

> Keeping the reference will result in the license file also being included in the package.

Is there anything wrong with it? I thought it was a best practice.
I'm not sure about .NET standards, I'll defer to @baywet and/or @andrueastman the final decision.

File: csharp/Std.UriTemplate.csproj [hunk 2]
```diff
@@ -19,7 +19,7 @@
     <RepositoryUrl>https://github.com/std-uritemplate/std-uritemplate</RepositoryUrl>
     <IncludeSymbols>true</IncludeSymbols>
     <SymbolPackageFormat>snupkg</SymbolPackageFormat>
-    <PackageLicenseFile>LICENSE</PackageLicenseFile>
+    <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
```
Comment (baywet) [inline csharp/Std.UriTemplate.csproj:22]: if the goal is to reduce the package size, the block line 53 should also be removed

**Review by baywet — CHANGES_REQUESTED** [review]
From reading [the specification](https://github.com/NuGet/Home/wiki/Packaging-License-within-the-nupkg), I couldn't find that one was necessarily better than the other.
A little history: license information used to be caried via a URL. This was bad for multiple reasons: people needed to click (time consuming, broken links, privacy, content change, etc...), and the nuget platform couldn't parse the information.
For that reason, [using a URL nowadays is discouraged](https://learn.microsoft.com/en-us/nuget/create-packages/package-authoring-best-practices#types-of-recommendations)
The nuget team introduce the file and the expression fields in an effort to collocate the license with the package and alleviate the URL issues.
From reading between the lines, I believe the expression has slight advantages:
- smaller package to download.
- less parsing effort from nuget.org to "understand" the license.
The main tradeoff I can see being:
- the actual content of the license on GitHub and the template from nuget referred to by the expression could differ.

I hope that helps making a choice either way.

**Comment by baywet** [comment]
If we want to diff our license file against the template, here is the source https://spdx.org/licenses/

File: csharp/Std.UriTemplate.csproj (hunk 2 above)
Comment (thompson-tomo) [reply csharp/Std.UriTemplate.csproj:22]: My goal is to facilitate license usage via SBOM'S however happy to remove the file if desired.

**Comment by baywet** [comment]
I just ran a diff, they are identical except for formatting and the fact that Red Hat is in the copyright (should it?)

**Comment by andreaTP** [comment]
> Red Hat is in the copyright (should it?)

I probably took it from another RH project, I'm fine removing the attribution.

**Comment by baywet** [comment]
> > Red Hat is in the copyright (should it?)
> 
> I probably took it from another RH project, I'm fine removing the attribution.

Since we made a point of creating a separate organization, project, etc... it'd make sense to align licensing as well.
Since we don't have a legal entity behind this project, I think our only option would be to set individuals.
And I don't want to "take credit for work you've mostly done", but for a "hit by a bus reason" I think it'd make sense to have multiple individuals here.
What do you think?

**Comment by andreaTP** [comment]
No strong opinions, I would be fine leaving it empty but feel free to proceed as you think is the best.

**Comment by andreaTP** [comment]
This should be the answer: https://www.apache.org/licenses/LICENSE-2.0#apply
To me it can be:

> Copyright 2023 std-uritemplate

Optionally with a link to: https://github.com/orgs/std-uritemplate/people
But, again, I'm really not fussy about this.

**Comment by baywet** [comment]
I think you just found the perfect solution!

**Comment by andreaTP** [comment]
@baywet do you mind to recap the next steps?

additional note: I'm extracting the C# implementation to a separate repo: https://github.com/std-uritemplate/std-uritemplate-csharp so I'll probably ask to direct this PR there sometime soon.

**Comment by baywet** [comment]
sure:
- [ ] remove the property group block that includes the license file from the csproj
- [ ] update the license file to set the copyright to `Copyright 2023 std-uritemplate`

CC @thompson-tomo

File: csharp/Std.UriTemplate.csproj (hunk 2 above)
Comment (thompson-tomo) [reply csharp/Std.UriTemplate.csproj:22]: File removed

**Comment by thompson-tomo** [comment]
@baywet ok, the changes requested are done.

**Review by andrueastman — APPROVED** [review]

**Comment by andreaTP** [comment]
Superseeded by #165 and https://github.com/std-uritemplate/std-uritemplate-csharp/pull/2

**Pull request closed without merging**